vllm/transformers_utils/tokenizer_group/base_tokenizer_group.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from transformers import PreTrainedTokenizer
+
+from vllm.lora.request import LoRARequest
+from vllm.utils import LRUCache
+from vllm.transformers_utils.tokenizer import get_tokenizer
+
+
+class BaseTokenizerGroup(ABC):
+
+    def __init__(self, tokenizer_id: str, enable_lora: bool, max_num_seqs: int,
+                 max_input_length: Optional[int], **tokenizer_config):
+        self.tokenizer_id = tokenizer_id
+        self.tokenizer_config = tokenizer_config
+        self.enable_lora = enable_lora
+        self.max_input_length = max_input_length
+        self.tokenizer = get_tokenizer(self.tokenizer_id, **tokenizer_config)
+        if enable_lora:
+            self.lora_tokenizers = LRUCache(capacity=max_num_seqs)
+        else:
+            self.lora_tokenizers = None
```

Comment:
There's some design smells here --
* Splitting implementation details between parent class and child class makes things harder to read for little benefit. Can we move these inits to the impl and leave the interface fully abstract?
* The RayTokenizerGroupPool subclasses the TokenizerGroup but doesn't seem to share any implementation details with it. If all we need is fast access to the tokenizers then we should have a tokenizer group inside of the RayTokenizerGroupPool instead of making the RayTokenizerGroupPool a tokenizer group itself.
* The distinction between Group and Pool will require every engineer to look it up when they first encounter it. We can add some docstrings to make it clear for them, but even better will be to remove the notion of pool altogether: `RayTokenizerGroup` -> `_RemoteTokenizerGroup` and then `RayTokenizerGroupPool` -> `RayTokenizerGroup`.

Seems to me we can have an interface for `TokenizerGroup` and then both the `BlockingTokenizerGroup` and `RayTokenizerGroup` can implement it, with the `RayTokenizerGroup` having its own `BlockingTokenizerGroup` for fast access and then remote `_RemoteTokenizerGroup = ray.remote(BlockingTokenizerGroup)` workers.